tests/test_appdb.py

```diff
@@ -705,9 +714,51 @@ async def test_unsupported_attribute(tmp_path, dev_init):
     assert "location_desc" in dev.endpoints[3].in_clusters[0].unsupported_attributes
     assert 0x0011 in dev.endpoints[3].in_clusters[0].unsupported_attributes
     assert "physical_env" in dev.endpoints[3].in_clusters[0].unsupported_attributes
-
     await app2.shutdown()
 
+    async def mockrequest(
+        is_general_req, command, schema, args, manufacturer=None, **kwargs
+    ):
+        assert is_general_req is True
+        assert command == 0
+        rar0010 = _mk_rar(0x0010, "Not Removed", zigpy.zcl.foundation.Status.SUCCESS)
+        return [[rar0010]]
+
+    # Now lets remove an unsupported attribute and make sure it is removed
+    app3 = await make_app(db)
+    dev = app3.get_device(ieee)
+    assert dev.is_initialized == dev_init
+    assert dev.endpoints[3].device_type == profiles.zha.DeviceType.PUMP
+    cluster = dev.endpoints[3].in_clusters[0]
+    assert 0x0010 in dev.endpoints[3].in_clusters[0].unsupported_attributes
+    cluster.request = mockrequest
+    await cluster.read_attributes([0x0010], allow_cache=False)
+    if dev_init:
+        assert 0x0010 not in dev.endpoints[3].in_clusters[0].unsupported_attributes
+        assert (
+            "location_desc"
+            not in dev.endpoints[3].in_clusters[0].unsupported_attributes
+        )
+    assert 0x0011 in dev.endpoints[3].in_clusters[0].unsupported_attributes
+    assert "physical_env" in dev.endpoints[3].in_clusters[0].unsupported_attributes
+    await app3.shutdown()
+
+    # Everything should've been saved - check that it re-loads
+    app4 = await make_app(db)
+    dev = app4.get_device(ieee)
+    assert dev.is_initialized == dev_init
+    assert dev.endpoints[3].device_type == profiles.zha.DeviceType.PUMP
+    if dev_init:
```

Comment:
not sure why this is needed but isn't for the existing test? I copied the existing structures to add the functionality and both listeners have this guard:

```
if not cluster.endpoint.device.is_initialized:
            return
```
but only the parts of the test that I added seem to trip up when initialized == False

I must be missing something stupid

Reply:
removed the guards and fixed the all success branch as discussed